src/Translator/doc/index.rst

```diff
@@ -0,0 +1,108 @@
+Symfony UX Translator
+=====================
+
+Symfony UX Translator is a Symfony bundle providing the same mechanism as `Symfony Translator`_
+in JavaScript with a TypeScript integration, in Symfony applications. It is part of `the Symfony UX initiative`_.
+
+The `ICU Message Format`_ is also supported.
+
+Installation
+------------
+
+Before you start, make sure you have `Symfony UX configured in your app`_.
+
+Install this bundle using Composer and Symfony Flex:
+
+.. code-block:: terminal
+
+    $ composer require symfony/ux-translator
+
+    # Don't forget to install the JavaScript dependencies as well and compile
+    $ npm install --force
+    $ npm run watch
+
+    # or use yarn
+    $ yarn install --force
+    $ yarn watch
+
+Usage
+-----
+
+When warming up the Symfony cache, all of your translations will be dumped as JavaScript into the ``var/translations/`` directory.
+For a better developer experience, TypeScript types definitions are also generated aside those JavaScript files.
+
+Then, you will be able to import those JavaScript translations in your assets.
+Don't worry about your final bundle size, only the translations you use will be included in your final bundle, thanks to the [tree shaking](https://webpack.js.org/guides/tree-shaking/).
+
+Configuring the default locale and locale fallbacks
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+By default, if the JavaScript translator is not configured, it will:
+- use the value of ``document.documentElement.lang`` or ``en`` (English) as the default locale
+- not use any locale fallbacks
```

Comment:
Should the fallback be `en` if the default isn't `en`?

Reply:
IMOthey are not related, or at least not here.

Reply:
IMO they are not related, or at least not here. 

The locale fallbacks (in this package) is represented as the following form: `{ locale: locale_fallback|null }`. 

If you have three locales `fr_FR`, `en_US` and `de_DE` in your apps, and the default locale is `en`, then this will be interpreted like this: 
```php
[
    'fr_FR' => 'fr',
    'fr' => 'en',
    'en' => null,
    'en_US' => 'en',
    'de_DE' => 'de',
    'de' => 'en',
]
```

But maybe I missed your point

Reply:
> not use any locale fallbacks

This is what threw me - I interpreted this is "no fallback will be used" so if the message key is `some.message`, the lang is `fr` and that key doesn't exist in the `fr` catalog, `some.message` would be rendered in js.

Reply:
Yes, the fallback locale will be used if it exist and a translation is present.

If this is not the case, we will render `some.message` like the Symfony's Translator do.

I've added a test case for that: 
```ts 
setLocaleFallbacks({'fr_FR':'fr','fr':'en','en_US':'en','en_GB':'en','de_DE':'de','de':'en'});

const MESSAGE_FRENCH_ONLY: Message<{ messages: { parameters: NoParametersType } }, 'fr'> = {
    id: 'message_french_only',
    translations: {
        messages: {
            fr: 'Un message en français uniquement',
        }
    }
}

expect(trans(MESSAGE_FRENCH_ONLY, {}, 'messages', 'fr')).toEqual('Un message en français uniquement');
expect(trans(MESSAGE_FRENCH_ONLY, {}, 'messages', 'en' as 'fr')).toEqual('message_french_only');
```